File: packages/sdk/python/compose_sdk/core/ui/types/selectOption.py
from typing import TypedDict, Union, List
from typing_extensions import NotRequired

SelectOptionValue = Union[str, int, bool]


class SelectOptionDict(TypedDict):
    value: SelectOptionValue
    label: str
    description: NotRequired[str]


SelectOptionPrimitive = SelectOptionValue

SelectOption = Union[SelectOptionDict, SelectOptionPrimitive]
SelectOptions = List[SelectOption]

File: packages/lib/ts-public/src/ui/types/selectOption.ts
type SelectOptionValue = string | number | boolean;

type SelectOptionDict = {
  value: SelectOptionValue;
  label: string;
  description?: string;
};

type SelectOptionPrimitive = SelectOptionValue;

type SelectOption = SelectOptionDict | SelectOptionValue;

type SelectOptions = readonly SelectOption[];

type ExtractOptionValue<TOption extends SelectOption> =
  TOption extends SelectOptionDict ? TOption["value"] : TOption;

export {
  type SelectOptionValue as Value,
  type SelectOptionDict as AsDict,
  type SelectOptionPrimitive as AsPrimitive,
  type SelectOption as Type,
  type SelectOptions as List,
  type ExtractOptionValue,
};
